crypto/ec/ec_key.c

```diff
@@ -292,6 +294,17 @@ int ec_generate_key(OPENSSL_CTX *libctx, EC_KEY *eckey, int pairwise_test)
      * 1 + rand[0..n-2] would effect the way that tests feed dummy entropy into
      * rand so the simpler backward compatible method has been used here.
      */
+
+    /* range of SM2 private key is [1, n-1) */
+    if (sm2) {
+        if (!BN_sub(order, tmp, BN_value_one()))
```

Comment:
But `order` is NULL here so won't this just crash?

Reply:
Ahh, this will definitely crash. Going to fix.